### ISSUE TITLE
Regression: "make depend" now required to build OpenSSL

### ISSUE THREAD

**Issue by quanah** [openssl/openssl#492]
```
making all in crypto/evp...
make[5]: Entering directory `/home/build/p4/zimbra/main/ThirdParty/openssl/tmp/UBUNTU12_64/zimbra-openssl/crypto/evp'
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o encode.o encode.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o digest.o digest.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o evp_enc.o evp_enc.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o evp_key.o evp_key.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o evp_acnf.o evp_acnf.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o evp_cnf.o evp_cnf.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o e_des.o e_des.c
gcc -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -g -O2 -DOPENSSL_NO_HEARTBEATS -m64 -DL_ENDIAN -O3 -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -D_FORTIFY_SOURCE=2  -c -o e_bf.o e_bf.c
make[5]: *** No rule to make target `../../include/openssl/idea.h', needed by `e_idea.o'.  Stop.
```

**Comment by quanah** [comment]
duplicate of #491

**Comment by quanah** [comment]
Not the same as #491.  "make depend" must now be run to build, which is a regression from prior releases.

**Comment by mattcaswell** [comment]
Please can you post your build steps. I don't get this behaviour, so I'm trying to figure out what's different for you.

**Comment by levitte** [comment]
As a side note, the three backquote syntax in Markdown require them to be on their own lines.  I just edited the starting text accordingly, and it immediately became more readable.

**Comment by mattcaswell** [comment]
Just seen your note on openssl-dev - so I replied there. Summary: you are configuring with "no-idea"; therefore you do need to run "make depend"; somehow you got away with it before :-)

**Comment by levitte** [comment]
@quanah  got away with it because the previous tarballs had all the symlinks (made as a result of `./config` without `no-idea`).

**Comment by quanah** [comment]
@mattcaswell Yeah, this worked on all prior 1.0.1, 1.0.0, and 0.9.8 builds I ever did.  So it may have been unintended that it did work, but it still comes out as a regression.  It may trip up others as well.

**Comment by quanah** [comment]
@levitte thanks for the cleanup!

**Comment by mattcaswell** [comment]
@quanah, agreed that this is less than ideal. This should correct itself in the next release when we sort out the problem that caused the screwed up sym links in the first place. I'll leave this issue open until we've done that. Its probably not a big enough issue in itself to reissue the tarballs (again).

**Comment by quanah** [comment]
@mattcaswell Agreed.  I've updated my debian rules to just run make depend, and am no longer blocked on our update for Zimbra.

**Comment by vszakats** [comment]
Want to second this. After suppressing the 'tar' failure (exit code 2), I've run into the same problem:
* Correct log with 1.0.2d:
https://ci.appveyor.com/project/vsz/harbour-deps/build/1.0.130
* Failure to build some parts with 1.0.2e:
https://ci.appveyor.com/project/vsz/harbour-deps/build/1.0.144

**Comment by vszakats** [comment]
Any suggestion how to fix it? I'm already doing a `make depend` [1], and unsure how to proceed from there.

[1] https://github.com/vszakats/harbour-deps/blob/test/mk-openssl.bat#L30

**Comment by levitte** [comment]
@vszakats, from looking at https://ci.appveyor.com/project/vsz/harbour-deps/build/1.0.144#L428 ,
it doesn't look like make depend was performed...

**Comment by vszakats** [comment]
@levitte Not sure what I'm missing, this is the sequence I'm executing since 1.0.2a [(now with echo enabled)](https://ci.appveyor.com/project/vsz/harbour-deps/build/1.0.150):

```batch
perl Configure mingw   shared no-unit-test no-ssl2 no-ssl3 no-rc5 no-idea no-dso no-sse2 "--prefix=%CD%"
sh -c mingw32-make depend
sh -c mingw32-make
```

Where should the `make depend` line go, to work as expected?
At the moment, it appears to start doing the build. [which BTW, fails in `tools/c_rehash`, and the second `make` call makes finish the build &mdash; just noticed that now.]

**Comment by levitte** [comment]
Well, the log lines I pointed at look like this:

    Since you've disabled or enabled at least one algorithm, you need to do
    the following before building:
     
    	make depend
     
    Configured for mingw.
    making all in crypto...
    mingw32-make[1]: Entering directory 'C:/w/openssl/crypto'
    C:/PROGRA~1/Git/usr/bin/perl.exe ../util/mkbuildinf.pl "gcc -I. -I.. -I../include  -D_WINDLL -DOPENSSL_USE_APPLINK -DOPENSSL_PIC -DOPENSSL_THREADS -D_MT -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -fomit-frame-pointer -O3 -march=i686 -mtune=generic -m32 -fno-ident -flto -ffat-lto-objects -frandom-seed=1449153871 -static-libgcc -Wall -DOPENSSL_BN_ASM_PART_WORDS -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DRMD160_ASM -DAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM" "mingw" >buildinf.h

Take note that the last lines from `Configure` is directly followed by `making all in crypto`.  *If* your `make depend` line actually performed, you should have seen something approximating to this:

    making depend in crypto...
    make[1]: Går till katalogen ”/home/levitte/gitwrk/openssl.net/official/1.0.2/crypto”
    making depend in crypto/objects...
    make[2]: Entering directory '/home/levitte/gitwrk/openssl.net/official/1.0.2/crypto/objects'
    ../../util/domd ../.. -MD x86_64-w64-mingw32-gcc -- -D_WINDLL -DOPENSSL_PIC -DOPENSSL_THREADS -D_MT -DDSO_WIN32 -Wno-pedantic-ms-format -DL_ENDIAN -O3 -Wall -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -DOPENSSL_NO_DEPRECATED -DOPENSSL_NO_EC_NISTP_64_GCC_128 -DOPENSSL_NO_GMP -DOPENSSL_NO_JPAKE -DOPENSSL_NO_LIBUNBOUND -DOPENSSL_NO_MD2 -DOPENSSL_NO_RC5 -DOPENSSL_NO_RFC3779 -DOPENSSL_NO_SCTP -DOPENSSL_NO_SSL_TRACE -DOPENSSL_NO_STORE -DOPENSSL_NO_UNIT_TEST --  o_names.c obj_dat.c obj_lib.c obj_err.c obj_xref.c

Note the starting line `making depend in crypto`...

If I were you, I'd have a closer look at what actually happens with your `make depend` line.

**Comment by vszakats** [comment]
I see and understand that I'm instructed to do a `make depend`, which is exactly what I'm executing right after `Configure`. But apparently, calling `make depend` just doesn't work:
```
C:\w\openssl>sh -c mingw32-make depend 
making all in crypto...
```
The output that follows gives no hint on why `depend` is not recognised and performed as instructed.
Any clues or past experience on why that happens, or where to look for more information?

**Comment by mattcaswell** [comment]
Shouldn't that be `sh -c "mingw32-make depend"`, i.e. with quotes?

**Comment by levitte** [comment]
I think @mattcaswell nailed it.

**Comment by vszakats** [comment]
Touché. Thanks a lot to you both! 👍🏻

**Comment by richsalz** [comment]
Okay, in 1.1 'make depend' is required and the issues brought up for other versions and platforms seem to have been addressed in the comments.  Closing this.
